addon/services/project-version.js

```diff
@@ -1,7 +1,36 @@
-import Service from '@ember/service';
+import Service, { inject as service } from '@ember/service';
+import { getOwner } from '@ember/application';
+import { resolve } from 'rsvp';
 
 export default Service.extend({
+  ajax: service(),
 
-  version: null
+  current: null,
+  root: null,
 
+  init() {
+    let { rootURL } = getOwner(this).resolveRegistration('config:environment');
+    let slash = rootURL.indexOf('/', 1);
+
+    // TODO deal with apps deployed to custom domains, so their pathnames don't have a leading
+    // segmenet for the project name. This will impact this service and the 404 page.
+    this.set('root', rootURL.slice(0, slash));
```

Comment:
If people use the default layout, we could, but there's technically nothing stopping you from returning something like `version/1.2.3` as the place you want a particular deploy to go, so we can't be certain the version is only a single segment.

That said, that approach actually feels better to me than what's currently here, and I almost built it that way until I realized that the 404 redirect logic _needs_ to know the number of static segments at the start in order to work. Since we can't get away from that, I decided to leave this as is so they could both pull from the same source when we do fix this.